### PR TITLE
Preload and persist Sprite Images

### PR DESCRIPTION
**Pull request by ajpal**
A couple changes here:
1. Load sprites as p5 Images, not Animations, for Sprite Lab. SpriteLab sprites are already forced to be single frame so this change doesn't actually change anything.
2. Persist p5 Images across runs. Similar idea to #29491. Only load an image if it hasn't been loaded yet (new selection from animation library or user-drawn image) or if the dataURI has changed (user edited an animation in piskel) 
3. Reset p5 on every blockspace change so that preview picks up new animations as soon as possible. Now that p5 preloading is much faster (since we are persisting unchanged images across runs), we can call do the whole p5 preload every time we refresh the preview. This allows us to get new animations as soon as possible so they show up in preview.

### CODE REVIEW

File: apps/src/gamelab/GameLabP5.js [hunk 1]
```diff
@@ -608,6 +608,41 @@ GameLabP5.prototype.preloadBackgrounds = function() {
...
+  if (!this.preloadedSprites) {
+    this.preloadedSprites = {};
+  }
+  this.preloadSpriteImages_ = Promise.all(
+    animationList.orderedKeys.map(key => {
+      const props = animationList.propsByKey[key];
+      if (
+        this.preloadedSprites[props.name] &&
+        this.preloadedSprites[props.name].dataURI === props.dataURI
+      ) {
```
Comment (jmkulwik) [inline apps/src/gamelab/GameLabP5.js]: Should we be doing the same thing for backgrounds?

Comment (ajpal) [reply apps/src/gamelab/GameLabP5.js]: Students cannot add or edit backgrounds, so preloading the set list on initial page load is sufficient. For sprites, students can add new sprites from the library and/or edit existing sprites at any time, so we need to check if anything has changed (and load any new assets) during each p5 preload phase.

**Comment by jmkulwik** [comment]
> ... Now that p5 preloading is much faster (since we are persisting unchanged images across runs), we can call do the whole p5 preload every time we refresh the preview...

Did you check that the p5 preload doesn't feel too slow on a slow connection/low RAM situation?

File: apps/src/gamelab/GameLabP5.js (hunk 1 above)
Comment (jmkulwik) [reply apps/src/gamelab/GameLabP5.js]: 👍

**Comment by ajpal** [comment]
Screen Shot 2019-08-05 at 11 35 52 AM

Performance profile looks very similar

**Review by jmkulwik — APPROVED** [review]

**Pull request merged**